src/node_osrm_support.hpp

```diff
@@ -424,9 +424,46 @@ inline bool parseCommonParameters(const v8::Local<v8::Object> &obj, ParamType &p
         {
             params->annotations = annotations->BooleanValue();
         }
+        else if (annotations->IsArray())
+        {
+            v8::Local<v8::Array> annotations_array = v8::Local<v8::Array>::Cast(annotations);
+            for (std::size_t i = 0; i < annotations_array->Length(); i++)
+            {
+                std::string annotations_str = *v8::String::Utf8Value(annotations_array->Get(i));
```

Comment:
Can you check for the ptr not being a nullptr?

Also please use Nan's string functions which are portable:
https://github.com/nodejs/nan/blob/master/doc/v8_misc.md

```c++
const Nan::Utf8String utf8str(source);
std::string s{*utf8str, *utf8str + utfstr.length()};